apps/aechannel/test/aesc_test_utils.erl

```diff
@@ -395,23 +395,23 @@ state_tx(ChannelPubKey, Initiator, Responder, Spec0) ->
                 aec_trees:hash(Trees);
             V -> V
         end,
-    OffChainSpec0
-        = #{channel_id         => aeser_id:create(channel, ChannelPubKey),
-            updates            => maps:get(updates, Spec, []),
-            state_hash         => StateHash,
-            round              => maps:get(round, Spec)},
     OffChainSpec =
-        case maps:get(block_hash, Spec0, none) of
-            none -> OffChainSpec0;
-            Val -> OffChainSpec0#{block_hash => Val}
-        end,
+        lists:foldl(
+            fun(Key, AccumSpec) ->
+                case maps:get(Key, Spec0, none) of
```

Comment:
I find `maps:find/2` more intuitive in these cases.

Reply:
Perhaps even better would be `maps:merge(#{channel_id => ...}, maps:with([block_hash, updates], Spec0))`